clients/src/main/java/org/apache/kafka/common/errors/RemoteStorageNotReadyException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.errors;
+
+/**
+ * An exception that indicates remote storage is not ready to receive the requests yet.
```

Comment:
Can you please expand on the documentation with what we discussed in KIP discussion thread please

Reply:
Updated the Javadoc. 


We cannot introduce a new error-code as the clients will throw `IllegalStateException` back to the caller when it encounters any new error-code. See [FetchCollector#handleInitializeErrors](https://sourcegraph.com/github.com/apache/kafka@trunk/-/blob/clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchCollector.java?L309). What options do we have?

With the existing `REPLICA_NOT_AVAILABLE` error, the client will request a metadata update which we want to avoid. It seems there is no other option left for us except `UNKNOWN_SERVER_ERROR` which is generic.

Reply:
If there are no alternatives, we may have to consider dropping the KIP or patch. This is something I should have brought up during the KIP discussion. I overlooked the implications on the client code.